src/autoloads/characters.gd

```diff
@@ -155,34 +158,92 @@ func _process(delta: float) -> void:
 		var positions: Dictionary = {}
 		for characterId in _characterResources:
 			positions[characterId] = _characterResources[characterId].getPosition()
-		rpc("_updateAllCharacterPositions", positions)
+		rpc("_updateAllCharacterPositions", positions, broadcastDataQueue)
+		broadcastDataQueue = []
 	## If client
 	elif Connections.isClient():
 		if not get_tree().get_network_unique_id() in _characterResources:
 			return
 		## Send own character position to server
 		_sendMyCharacterPosToServer()
+	else:
+		assert(false, "Unreachable")
+
+## --Client functions
+
+func requestCharacterData() -> void:
+	rpc_id(1, "sendAllCharacterData")
 
 # puppet keyword means that when this function is used in an rpc call
 # 	it will only be run on client
-puppet func _updateAllCharacterPositions(positions: Dictionary) -> void:
+puppet func _updateAllCharacterPositions(positions: Dictionary, characterData: Array) -> void:
```

Comment:
If this function is also sending character data it should really be renamed to `_updateAllCharacterData()` or something along those lines